OpenTyping/InternetServices.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.IO;
+using System.Net;
+using System.Net.NetworkInformation;
+
+namespace OpenTyping
+{
+    class InternetServices
+    {
+        public static bool IsConnected()
```

Comment:
You can send the error reporting directly to the server without ping. At the try, you can check and separate the root cause of network issue by the returned error in try...catch. You can refer to the code on the globalrank-restful branch, https://github.com/roboticsware/OpenTyping/blob/6f456e947a0045a27a28cc078350ed632c72cca8/OpenTyping/RankServer.cs#L28.